cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/monitor/InternalRouterHandler.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.runtime.monitor;
+
+import com.google.inject.Inject;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.http.BodyConsumer;
+import io.cdap.http.HttpResponder;
+import io.netty.handler.codec.http.HttpRequest;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+
+/**
+ * The http handler for routing CDAP service requests from workers running in
+ * the same cluster. This service delegates all auth checks to downstream
+ * services.
+ */
+@Path(Constants.Gateway.INTERNAL_API_VERSION_3 + "/router")
+public class InternalRouterHandler extends AbstractServiceRoutingHandler {
+
+  @Inject
+  InternalRouterHandler(DiscoveryServiceClient discoveryServiceClient) {
+    super(discoveryServiceClient);
+  }
+
+  /**
+   * Handles GET and DELETE calls from program runtime to access CDAP services.
+   * It simply forwards the call to internal CDAP service.
+   */
+  @Path("/services/{service}/**")
+  @GET
+  @DELETE
+  public void routeService(HttpRequest request, HttpResponder responder,
+      @PathParam("service") String service) throws Exception {
+    routeService(request, responder, service, getServicePath(request, service));
+  }
+
+  /**
+   * Handles PUT and POST calls to access CDAP services. It simply forwards the
```

Comment:
Instead of tying them to http methods, write the functionality being done by the method, ie. consuming body and then returning response.

Reply:
Makes sense. Rewrote the java doc.